### ISSUE TITLE
Scary "failed to unmarshal" logs seen in journal on a k8s scale run

### ISSUE THREAD

**Issue by fasaxc** [projectcalico/calico-cni#237]
I'm not sure if these are benign or not; the "exit status 1" it hints that the CNI plugin is failing.  Scale run was Frank's 2.0 run on 2016-12-08.  I was digging into the diags to try to figure out why we saw poor TTFPs and found these logs.  Not sure if they're related.

```
Dec 08 17:48:57 tigera-scale-fd-003.c.unique-caldron-775.internal kubelet[1493]: Calico CNI checking for existing endpoint: <nil>
Dec 08 17:48:57 tigera-scale-fd-003.c.unique-caldron-775.internal kubelet[1493]: with error: exit status 1
Dec 08 17:48:57 tigera-scale-fd-003.c.unique-caldron-775.internal kubelet[1493]: Calico CNI IPAM request count IPv4=1 IPv6=0
Dec 08 17:48:57 tigera-scale-fd-003.c.unique-caldron-775.internal kubelet[1493]: time="2016-12-08T17:48:57Z" level=warning msg="Failed to unmarshal \"\" into value &model.BlockAffinity{}"
Dec 08 17:48:58 tigera-scale-fd-003.c.unique-caldron-775.internal kubelet[1493]: with error: exit status 1
```

**Comment by caseydavenport** [comment]
@fasaxc did you happen to notice how commonly this was occuring? Was it isolated to one node / time within the test?

Could you check explicitly and make a note of whether or not these show up in the test you + @frnkdny + @lwr20 are running tomorrow :)

**Comment by hanikesn** [comment]
I also see this error on our setup. Running calico-cni 1.4.2 .
I'm also getting the following errors during creation:
```
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: I1213 15:33:36.590661   17653 docker_manager.go:2162] Determined pod ip after infra change: "api-current-sprint-rel-113761345-usy73_cms(82e7b4a5-c149-11e6-9fb0-069d86b508b9)": "10.130.27.204"
Dec 13 15:33:36 ip-10-128-16-121 kernel: IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec 13 15:33:36 ip-10-128-16-121 kernel: IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: Calico CNI using IPs: [{{10.130.27.204 ffffffff}}]
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: Calico CNI IPAM assigned addresses IPv4=[10.130.27.204] IPv6=[]
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: ERROR: logging before flag.Parse: I1213 15:33:36.458393   26288 keys.go:225] Failed to unmarshal "" into value &model.BlockAffinity{}
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: ERROR: logging before flag.Parse: I1213 15:33:36.458371   26288 keys.go:225] Failed to unmarshal "" into value &model.BlockAffinity{}
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: ERROR: logging before flag.Parse: I1213 15:33:36.458205   26288 keys.go:225] Failed to unmarshal "" into value &model.BlockAffinity{}
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: Calico CNI IPAM request count IPv4=1 IPv6=0
Dec 13 15:33:36 ip-10-128-16-121 kubelet[17653]: Calico CNI checking for existing endpoint: <nil>
```

```
Dec 13 15:43:26 ip-10-128-16-121 kubelet[17653]: E1213 15:43:26.866983   17653 kubelet.go:2098] Failed killing the pod "api-current-sprint-rel-113761345-usy73": failed to "TeardownNetwork" for "api-current-sprint-rel-113761345-usy73_cms" with TeardownNetworkError: "Failed to teardown network for pod \"82e7b4a5-c149-11e6-9fb0-069d86b508b9\" using network plugins \"cni\": resource does not exist: WorkloadEndpoint(hostname=ip-10-128-16-121, orchestrator=k8s, workload=cms.api-current-sprint-rel-113761345-usy73, name=)"
Dec 13 15:43:26 ip-10-128-16-121 kubelet[17653]: E1213 15:43:26.864776   17653 docker_manager.go:1419] Failed to teardown network for pod "82e7b4a5-c149-11e6-9fb0-069d86b508b9" using network plugins "cni": resource does not exist: WorkloadEndpoint(hostname=ip-10-128-16-121, orchestrator=k8s, workload=cms.api-current-sprint-rel-113761345-usy73, name=)
Dec 13 15:43:26 ip-10-128-16-121 kubelet[17653]: E1213 15:43:26.864730   17653 cni.go:273] Error deleting network: resource does not exist: WorkloadEndpoint(hostname=ip-10-128-16-121, orchestrator=k8s, workload=cms.api-current-sprint-rel-113761345-usy73, name=)
Dec 13 15:43:26 ip-10-128-16-121 kubelet[17653]: time="2016-12-13T15:43:26Z" level=error msg="resource does not exist: {cms.api-current-sprint-rel-113761345-usy73}" WorkloadID=cms.api-current-sprint-rel-113761345-usy73
Dec 13 15:43:26 ip-10-128-16-121 kubelet[17653]: Calico CNI releasing IP address
```

**Comment by caseydavenport** [comment]
We got to the bottom of this the other day.  It's an incompatibility between the way libcalico-go expects affinity to be marked vs how the python code writes it.

There's still a tiny bit of python code writing this out when `ipip` is enabled.

@robbrockbank fixed this in https://github.com/projectcalico/libcalico-go/pull/301 and it will be included in Calico v2.0.0.

**Comment by tomdee** [comment]
@caseydavenport So is this fixed in the 1.5.4 release we just did?

**Comment by caseydavenport** [comment]
Yes, closing.
